src/domain/email/email.repository.ts

```diff
@@ -170,6 +171,19 @@ export class EmailRepository implements IEmailRepository {
     const newEmail = new EmailAddress(args.emailAddress);
     const currentEmail = await this.emailDataSource.getEmail(args);
 
+    if (
+      currentEmail.verificationGeneratedOn &&
+      this._isEmailVerificationCodeValid(currentEmail)
```

Comment:
Not sure I get the purpose of this condition. The error seems to be thrown when the currentEmail has a previously generated verificationCode, and that verificationCode is valid, is that correct? Maybe I'm missing something.

Reply:
> Not sure I get the purpose of this condition. The error seems to be thrown when the currentEmail has a previously generated verificationCode, and that verificationCode is valid, is that correct?

Yes, this prevents editing again within the TTL window as the `verificationCode` is only valid within it.

Reply:
Oh, you are right, I was misinterpreting the requirement 😅 Thanks!